UNITTESTS/stubs/CMakeLists.txt

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) 2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+add_library(mbed-stubs-headers INTERFACE)
+add_library(mbed-headers INTERFACE)
+add_library(mbed-headers-base INTERFACE)
+add_library(mbed-headers-platform INTERFACE)
+add_library(mbed-headers-connectivity INTERFACE)
+add_library(mbed-headers-storage INTERFACE)
+add_library(mbed-headers-drivers INTERFACE)
+add_library(mbed-headers-hal INTERFACE)
+add_library(mbed-headers-events INTERFACE)
+add_library(mbed-headers-rtos INTERFACE)
+
+target_link_libraries(mbed-headers
+    INTERFACE
+        mbed-headers-base
+        mbed-headers-platform
+        mbed-headers-connectivity
+        mbed-headers-storage
+        mbed-headers-drivers
+        mbed-headers-hal
+        mbed-headers-events
+        mbed-headers-rtos
+)
+
+target_include_directories(mbed-headers-platform
```

Comment:
This feels like duplication to me - as these are "real" headers, can't we define them directly in the relevant directory's `CMakeLists.txt` (for instance `/platform/CMakeLists.txt`? This goes quite against any future componentisation effort. The "real" targets can then link to these "headers" targets.

Reply:
The upcoming task in the backlog to create Refactor components for independent unit testing as part of that we will do the further cleanup in `UNITTESTS/stubs`, `UNITTESTS/target_h` to move this to the respective components directory. For example, we will move the platform stubs and header library creation inside `mbed-os/platform/tests/UNITTESTS` directory

Reply:
Any reason why that work is not made as part of this PR?

Reply:
As the first set of work planned to modify existing unit test CMake support to proper CMake way of doing things and further next phase of work will plan moving stubs/target_h closure to library

Reply:
@rajkan01 Can you create tracking issue for this planned work?

Reply:
> 
> 
> @rajkan01 Can you create tracking issue for this planned work?

@0xc0170 Already raised Github issues for this work